specification/keyvault/Security.KeyVault.BackupRestore/routes.tsp

```diff
@@ -0,0 +1,256 @@
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-client-generator-core";
+import "@typespec/rest";
+import "@typespec/versioning";
+import "./models.tsp";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Azure.Core;
+using Azure.ClientGenerator.Core;
+
+namespace KeyVault;
+
+/**
+ * Returns the status of full backup operation
+ */
+#suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Foundations.Operation is necessary for Key Vault"
+#suppress "@azure-tools/typespec-azure-core/use-standard-names" "Operation name is already established"
+@route("/backup/{jobId}/pending")
+@get
+op fullBackupStatus is Azure.Core.Foundations.Operation<
+  {
+    /**
+     * The id returned as part of the backup request
+     */
+    @path
+    jobId: string;
+  },
+  FullBackupOperation,
+  {},
```

Comment:
I would make this more dry by creating a custom partially specialized template so I don't have to remember to provide `KeyVaultError` everywhere.

Reply:
Done in https://github.com/Azure/azure-rest-api-specs/pull/30239/commits/391e5b8505f168e7638ae392452cb0269bceb787 -- thanks for the suggestion!